From 9fedc6f999cbdc2d3e7da1a0a7dda8c76b3ec59f Mon Sep 17 00:00:00 2001
Message-Id: <9fedc6f999cbdc2d3e7da1a0a7dda8c76b3ec59f.1426052288.git.chang-joon.lee@intel.com>
In-Reply-To: <99f2c84a457bac958678aa60ef795f22948aa382.1426052288.git.chang-joon.lee@intel.com>
References: <99f2c84a457bac958678aa60ef795f22948aa382.1426052288.git.chang-joon.lee@intel.com>
From: Durgadoss R <durgadoss.r@intel.com>
Date: Tue, 30 Dec 2014 14:40:00 +0530
Subject: [PATCH 3/5] REVERTME [VPG]: drm/i915: Skip power sequencer kick if
 port is active

As part of the power sequencer kick method, we disable a port.
This works if the port is not active; Instead if it is, this
causes FIFO underruns and hence display appears to be blank.

Fix this by skipping the power sequencer if the port is already
enabled. This fix is a part of an upstream patch as well, whose
details are mentioned below. Once the upstream patch is pulled in,
we can revert this patch.
Upstream-repo: git://anongit.freedesktop.org/drm-intel
Upstream-commit: 961a0db00977ff98ff99d942ae4c046423e1aebc

Change-Id: I2f2ce33886ad82dadc47436bd7e07027570cf22c
Issue: BAP-504
Signed-off-by: Durgadoss R <durgadoss.r@intel.com>
---
 drivers/gpu/drm/i915/intel_dp.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index 438e272..a1527b4 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -303,6 +303,12 @@ vlv_power_sequencer_kick(struct intel_dp *intel_dp,
 	enum intel_display_power_domain power_domain;
 	uint32_t DP;
 
+	if (I915_READ(intel_dp->output_reg) & DP_PORT_EN) {
+		DRM_DEBUG_DRIVER(
+		"Skip power sequencer kick since port is active\n");
+		return;
+	}
+
 	/* Preserve the BIOS-computed detected bit. This is
 	 * supposed to be read-only.
 	 */
-- 
1.7.9.5

